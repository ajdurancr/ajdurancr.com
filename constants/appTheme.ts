const DEFAULT = 'default'
const DARK = 'dark'
const LIGHT = 'light'
const GOLD = 'gold'

export const imageSuffixByTheme = {
  [DEFAULT]: '-light',
  [LIGHT]: '',
  [DARK]: '-light',
}

export const themeOptions = Object.freeze({
  DEFAULT,
  DARK,
  GOLD,
  LIGHT,
})
