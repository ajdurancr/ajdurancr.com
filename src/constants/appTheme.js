const DEFAULT = 'default'
const DARK = 'dark'
const LIGHT = 'light'
const GOLD = 'gold'

export const imageSuffixByTheme = {
  [DEFAULT]: '-light',
  [LIGHT]: '',
  [DARK]: '-light',
}

export default {
  DEFAULT,
  DARK,
  GOLD,
  LIGHT,
}
