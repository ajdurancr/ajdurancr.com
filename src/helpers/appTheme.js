import get from 'lodash.get'
import appThemeMap, { imageSuffixByTheme } from '../constants/appTheme'

export const getImageThemeSuffix = (appTheme, customSuffixes = {}) => get(
  { ...imageSuffixByTheme, ...customSuffixes },
  `${appTheme}`,
  imageSuffixByTheme[appThemeMap.DEFAULT],
)
