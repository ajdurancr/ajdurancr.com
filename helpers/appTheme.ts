import get from 'lodash.get'

import { imageSuffixByTheme, themeOptions } from '../constants/appTheme'

export const getImageThemeSuffix = (appTheme: string, customSuffixes = {}) => get(
  { ...imageSuffixByTheme, ...customSuffixes },
  `${appTheme}`,
  imageSuffixByTheme[themeOptions.DEFAULT],
)
