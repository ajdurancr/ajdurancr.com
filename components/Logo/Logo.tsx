import React from 'react'
import classnames from 'classnames'

import { Image } from '../Image'
import { themeOptions } from '../../constants/appTheme'
import { getImageThemeSuffix } from '../../helpers/appTheme'
import useAppTheme from '../../hooks/useAppTheme'
import styles from './Logo.module.sass'

const Logo: React.FC<{ className?: string, onlyIcon?: boolean }> = ({ className = '', onlyIcon }) => {
  const appTheme = useAppTheme()
  const rootUrl = onlyIcon ? 'icon' : 'logo'
  const imageSuffix = getImageThemeSuffix(appTheme, { [themeOptions.GOLD]: '-gold' })

  return (
    <Image
      className={classnames(styles.container, className)}
      src={`/images/logo/${rootUrl}${imageSuffix}.svg`}
      alt={`${rootUrl}${imageSuffix.replace('-', ' ')}`}
    />

  )
}

export { Logo }
