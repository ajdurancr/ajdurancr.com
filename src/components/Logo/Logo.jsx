import React from 'react'
import propTypes from 'prop-types'
import classnames from 'classnames'

import { Image } from '..'
import appThemeMap from '../../constants/appTheme'
import { getImageThemeSuffix } from '../../helpers/appTheme'
import useAppTheme from '../../hooks/useAppTheme'
import styles from './Logo.module.sass'

const Logo = ({ className, onlyIcon }) => {
  const appTheme = useAppTheme()
  const rootUrl = onlyIcon ? 'icon' : 'logo'
  const imageSuffix = getImageThemeSuffix(appTheme, { [appThemeMap.GOLD]: '-gold' })

  return (
    <Image
      className={classnames(styles.container, className)}
      src={`/images/logo/${rootUrl}${imageSuffix}.svg`}
      alt={`${rootUrl}${imageSuffix.replace('-', ' ')}`}
    />

  )
}

Logo.defaultProps = {
  className: '',
  onlyIcon: false,
}

Logo.propTypes = {
  className: propTypes.string,
  onlyIcon: propTypes.bool,
}

export default Logo
