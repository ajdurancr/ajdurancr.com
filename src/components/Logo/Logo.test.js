import React from 'react'
import { shallow } from 'enzyme'

import Logo from './Logo'
import appTheme from '../../constants/appTheme'

jest.mock('../../hooks/useAppTheme', () => jest.fn())

const useAppTheme = require('../../hooks/useAppTheme')

describe('Logo', () => {
  beforeEach(() => {
    useAppTheme.mockReturnValue(appTheme.DEFAULT)
  })

  it('renders correctly default logo', () => {
    const wrapper = shallow(<Logo />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly, theme:dark', () => {
    useAppTheme.mockReturnValue(appTheme.DARK)
    const wrapper = shallow(<Logo />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly, theme:light', () => {
    useAppTheme.mockReturnValue(appTheme.LIGHT)
    const wrapper = shallow(<Logo />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly, theme:gold', () => {
    useAppTheme.mockReturnValue(appTheme.GOLD)
    const wrapper = shallow(<Logo />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly, onlyIcon with default logo', () => {
    const wrapper = shallow(<Logo onlyIcon />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly, onlyIcon with theme:dark', () => {
    useAppTheme.mockReturnValue(appTheme.DARK)
    const wrapper = shallow(<Logo onlyIcon />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly, onlyIcon with theme:light', () => {
    useAppTheme.mockReturnValue(appTheme.LIGHT)
    const wrapper = shallow(<Logo onlyIcon />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with className value', () => {
    const wrapper = shallow(<Logo className="testClassName" />)

    expect(wrapper).toMatchSnapshot()
  })
})
