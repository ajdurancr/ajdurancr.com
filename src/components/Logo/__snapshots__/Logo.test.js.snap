// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly default logo 1`] = `
<Image
  alt="logo light"
  className="container"
  src="/images/logo/logo-light.svg"
/>
`;

exports[`Logo renders correctly with className value 1`] = `
<Image
  alt="logo light"
  className="container testClassName"
  src="/images/logo/logo-light.svg"
/>
`;

exports[`Logo renders correctly, onlyIcon with default logo 1`] = `
<Image
  alt="icon light"
  className="container"
  src="/images/logo/icon-light.svg"
/>
`;

exports[`Logo renders correctly, onlyIcon with theme:dark 1`] = `
<Image
  alt="icon light"
  className="container"
  src="/images/logo/icon-light.svg"
/>
`;

exports[`Logo renders correctly, onlyIcon with theme:light 1`] = `
<Image
  alt="icon"
  className="container"
  src="/images/logo/icon.svg"
/>
`;

exports[`Logo renders correctly, theme:dark 1`] = `
<Image
  alt="logo light"
  className="container"
  src="/images/logo/logo-light.svg"
/>
`;

exports[`Logo renders correctly, theme:gold 1`] = `
<Image
  alt="logo gold"
  className="container"
  src="/images/logo/logo-gold.svg"
/>
`;

exports[`Logo renders correctly, theme:light 1`] = `
<Image
  alt="logo"
  className="container"
  src="/images/logo/logo.svg"
/>
`;
